### ISSUE TITLE
Incompatibility when using doorkeeper-mongodb 5.0 with doorkeeper 5.1.0

### ISSUE THREAD

**Issue by gsmendoza** [doorkeeper-gem/doorkeeper-mongodb#37]
Hi! I'm encountering the following error when using doorkeeper-mongodb 5.0 with doorkeeper 5.1.0:

### Steps to reproduce / Expected behavior

Given my gem versions are

* doorkeeper (5.1.0)
* doorkeeper-mongodb (5.0.0)

And I have a Doorkeeper application
When I go to /oauth/applications/:application_id
Then I should be able to view the Doorkeeper application

### Actual behavior

I get the following error:

```
 Started GET "/oauth/applications/5cdf7a58321dcb0007955ee0" for 172.18.0.1 at 2019-05-21 22:48:15 +0000
 Cannot render console from 172.18.0.1! Allowed networks: 127.0.0.1, ::1, 127.0.0.0/127.255.255.255
 Processing by Doorkeeper::ApplicationsController#show as HTML
   Parameters: {"id"=>"5cdf7a58321dcb0007955ee0"}
   Rendering /usr/local/bundle/gems/doorkeeper-5.1.0views/doorkeeper/applications/show.html.erb within layouts/doorkeeper/admin
   Rendered /usr/local/bundle/gems/doorkeeper-5.1.0views/doorkeeper/applications/show.html.erb within layouts/doorkeeper/admin (4.5ms)
 Completed 500 Internal Server Error in 18ms
 
 
   
 ActionView::Template::Error (undefined method `plaintext_secret' for #<Doorkeeper::Application:0x00007f50cdc82178>):
      8:     <p><code class="bg-light" id="application_id"><%= @application.uid %></code></p>
      9: 
     10:     <h4><%= t('.secret') %>:</h4>
     11:     <p><code class="bg-light" id="secret"><%= @application.plaintext_secret %></code></p>
     12: 
     13:     <h4><%= t('.scopes') %>:</h4>
     14:     <p><code class="bg-light" id="scopes"><%= @application.scopes.presence || raw('&nbsp;') %></code></p>
   
 doorkeeper (5.1.0views) doorkeeper/applications/show.html.erb:11:in `__usr_local_bundle_gems_doorkeeper______views_doorkeeper_applications_show_html_erb__3282679595179731542_69992513242040'
```

### Workaround

Roll back Doorkeeper to 5.0.2.

#### Ruby version

ruby 2.5.3p105 (2018-10-18 revision 65156) [x86_64-linux]

#### Gemfile.lock

```
GIT
  remote: https://github.com/SnackNation/json-logic-ruby.git
  revision: 52a84fdc4a84b85b4619d21e9a403889044d8815
  branch: deep_fetch_false_value
  specs:
    json_logic (0.4.1)

GIT
  remote: https://github.com/SnackNation/spring-watcher-listen.git
  revision: 2af02a92bef8a2e51d6601e40270228a3b3b9967
  specs:
    spring-watcher-listen (2.0.2)
      listen (>= 2.7, < 4.0)
      spring (>= 1.2, < 3.0)

GEM
  remote: https://rubygems.org/
  specs:
    actioncable (5.2.1)
      actionpack (= 5.2.1)
      nio4r (~> 2.0)
      websocket-driver (>= 0.6.1)
    actionmailer (5.2.1)
      actionpack (= 5.2.1)
      actionview (= 5.2.1)
      activejob (= 5.2.1)
      mail (~> 2.5, >= 2.5.4)
      rails-dom-testing (~> 2.0)
    actionpack (5.2.1)
      actionview (= 5.2.1)
      activesupport (= 5.2.1)
      rack (~> 2.0)
      rack-test (>= 0.6.3)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.0.2)
    actionview (5.2.1)
      activesupport (= 5.2.1)
      builder (~> 3.1)
      erubi (~> 1.4)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.0.3)
    activejob (5.2.1)
      activesupport (= 5.2.1)
      globalid (>= 0.3.6)
    activemodel (5.2.1)
      activesupport (= 5.2.1)
    activemodel-serializers-xml (1.0.2)
      activemodel (> 5.x)
      activesupport (> 5.x)
      builder (~> 3.1)
    activerecord (5.2.1)
      activemodel (= 5.2.1)
      activesupport (= 5.2.1)
      arel (>= 9.0)
    activestorage (5.2.1)
      actionpack (= 5.2.1)
      activerecord (= 5.2.1)
      marcel (~> 0.3.1)
    activesupport (5.2.1)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (>= 0.7, < 2)
      minitest (~> 5.1)
      tzinfo (~> 1.1)
    addressable (2.5.2)
      public_suffix (>= 2.0.2, < 4.0)
    arel (9.0.0)
    ast (2.4.0)
    attr_extras (5.2.0)
    bindex (0.5.0)
    bootsnap (1.3.1)
      msgpack (~> 1.0)
    brakeman (4.3.1)
    bson (4.4.2)
    builder (3.2.3)
    byebug (10.0.2)
    capybara (3.6.0)
      addressable
      mini_mime (>= 0.1.3)
      nokogiri (~> 1.8)
      rack (>= 1.6.0)
      rack-test (>= 0.6.3)
      xpath (~> 3.1)
    capybara-screenshot (1.0.21)
      capybara (>= 1.0, < 4)
      launchy
    childprocess (0.9.0)
      ffi (~> 1.0, >= 1.0.11)
    concurrent-ruby (1.1.4)
    config (1.7.1)
      activesupport (>= 3.0)
      deep_merge (~> 1.2.1)
      dry-validation (>= 0.12.2)
    crack (0.4.3)
      safe_yaml (~> 1.0.0)
    crass (1.0.4)
    database_cleaner (1.7.0)
    debase (0.2.2)
      debase-ruby_core_source (>= 0.10.2)
    debase-ruby_core_source (0.10.3)
    deep_merge (1.2.1)
    diff-lcs (1.3)
    doorkeeper (5.1.0)
      railties (>= 5)
    doorkeeper-mongodb (5.0.0)
      doorkeeper (>= 5.0, < 6.0)
    dotenv (2.5.0)
    dotenv-rails (2.5.0)
      dotenv (= 2.5.0)
      railties (>= 3.2, < 6.0)
    draper (3.0.1)
      actionpack (~> 5.0)
      activemodel (~> 5.0)
      activemodel-serializers-xml (~> 1.0)
      activesupport (~> 5.0)
      request_store (~> 1.0)
    dry-configurable (0.8.1)
      concurrent-ruby (~> 1.0)
      dry-core (~> 0.4, >= 0.4.7)
    dry-container (0.7.0)
      concurrent-ruby (~> 1.0)
      dry-configurable (~> 0.1, >= 0.1.3)
    dry-core (0.4.7)
      concurrent-ruby (~> 1.0)
    dry-equalizer (0.2.1)
    dry-inflector (0.1.2)
    dry-logic (0.5.0)
      dry-container (~> 0.2, >= 0.2.6)
      dry-core (~> 0.2)
      dry-equalizer (~> 0.2)
    dry-types (0.14.0)
      concurrent-ruby (~> 1.0)
      dry-container (~> 0.3)
      dry-core (~> 0.4, >= 0.4.4)
      dry-equalizer (~> 0.2)
      dry-inflector (~> 0.1, >= 0.1.2)
      dry-logic (~> 0.5, >= 0.5)
    dry-validation (0.13.0)
      concurrent-ruby (~> 1.0)
      dry-configurable (~> 0.1, >= 0.1.3)
      dry-core (~> 0.2, >= 0.2.1)
      dry-equalizer (~> 0.2)
      dry-logic (~> 0.5, >= 0.5.0)
      dry-types (~> 0.14, >= 0.14)
    erubi (1.7.1)
    execjs (2.7.0)
    factory_bot (4.11.1)
      activesupport (>= 3.0.0)
    factory_bot_rails (4.11.1)
      factory_bot (~> 4.11.1)
      railties (>= 3.0.0)
    faraday (0.12.2)
      multipart-post (>= 1.2, < 3)
    faraday_middleware (0.12.2)
      faraday (>= 0.7.4, < 1.0)
    ffi (1.9.25)
    globalid (0.4.1)
      activesupport (>= 4.2.0)
    hashdiff (0.3.7)
    hashie (3.5.7)
    i18n (1.5.3)
      concurrent-ruby (~> 1.0)
    jaro_winkler (1.5.1)
    jbuilder (2.7.0)
      activesupport (>= 4.2.0)
      multi_json (>= 1.2)
    jwt (1.5.6)
    kaminari-actionview (1.1.1)
      actionview
      kaminari-core (= 1.1.1)
    kaminari-core (1.1.1)
    kaminari-mongoid (1.0.1)
      kaminari-core (~> 1.0)
      mongoid
    launchy (2.4.3)
      addressable (~> 2.3)
    listen (3.1.5)
      rb-fsevent (~> 0.9, >= 0.9.4)
      rb-inotify (~> 0.9, >= 0.9.7)
      ruby_dep (~> 1.2)
    local_time (2.1.0)
    loofah (2.2.2)
      crass (~> 1.0.2)
      nokogiri (>= 1.5.9)
    mail (2.7.0)
      mini_mime (>= 0.1.1)
    marcel (0.3.2)
      mimemagic (~> 0.3.2)
    method_source (0.9.0)
    mimemagic (0.3.2)
    mini_mime (1.0.1)
    mini_portile2 (2.3.0)
    minitest (5.11.3)
    minitest-focus (1.1.2)
      minitest (>= 4, < 6)
    minitest-test_profile (0.2.2)
      minitest
    mongo (2.7.0)
      bson (>= 4.4.2, < 5.0.0)
    mongoid (7.0.2)
      activemodel (>= 5.1, < 6.0.0)
      mongo (>= 2.5.1, < 3.0.0)
    mongoid_rails_migrations (1.1.1)
      activesupport (>= 4.2.0)
      bundler (>= 1.0.0)
      mongoid (>= 4.0.0)
      rails (>= 4.2.0)
      railties (>= 4.2.0)
    msgpack (1.2.4)
    multi_json (1.13.1)
    multi_xml (0.6.0)
    multipart-post (2.0.0)
    nio4r (2.3.1)
    nokogiri (1.8.4)
      mini_portile2 (~> 2.3.0)
    oauth2 (1.4.0)
      faraday (>= 0.8, < 0.13)
      jwt (~> 1.0)
      multi_json (~> 1.3)
      multi_xml (~> 0.5)
      rack (>= 1.2, < 3)
    omniauth (1.8.1)
      hashie (>= 3.4.6, < 3.6.0)
      rack (>= 1.6.2, < 3)
    omniauth-google-oauth2 (0.5.3)
      jwt (>= 1.5)
      omniauth (>= 1.1.1)
      omniauth-oauth2 (>= 1.5)
    omniauth-oauth2 (1.5.0)
      oauth2 (~> 1.1)
      omniauth (~> 1.2)
    parallel (1.12.1)
    parser (2.5.1.2)
      ast (~> 2.4.0)
    powerpack (0.1.2)
    public_suffix (3.0.3)
    puma (3.12.0)
    rack (2.0.5)
    rack-proxy (0.6.5)
      rack
    rack-test (1.1.0)
      rack (>= 1.0, < 3)
    rails (5.2.1)
      actioncable (= 5.2.1)
      actionmailer (= 5.2.1)
      actionpack (= 5.2.1)
      actionview (= 5.2.1)
      activejob (= 5.2.1)
      activemodel (= 5.2.1)
      activerecord (= 5.2.1)
      activestorage (= 5.2.1)
      activesupport (= 5.2.1)
      bundler (>= 1.3.0)
      railties (= 5.2.1)
      sprockets-rails (>= 2.0.0)
    rails-controller-testing (1.0.2)
      actionpack (~> 5.x, >= 5.0.1)
      actionview (~> 5.x, >= 5.0.1)
      activesupport (~> 5.x)
    rails-dom-testing (2.0.3)
      activesupport (>= 4.2.0)
      nokogiri (>= 1.6)
    rails-html-sanitizer (1.0.4)
      loofah (~> 2.2, >= 2.2.2)
    railties (5.2.1)
      actionpack (= 5.2.1)
      activesupport (= 5.2.1)
      method_source
      rake (>= 0.8.7)
      thor (>= 0.19.0, < 2.0)
    rainbow (3.0.0)
    rake (12.3.1)
    rb-fsevent (0.10.3)
    rb-inotify (0.9.10)
      ffi (>= 0.5.0, < 2)
    request_store (1.4.1)
      rack (>= 1.4)
    rspec-mocks (3.8.0)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.8.0)
    rspec-support (3.8.0)
    rubocop (0.59.2)
      jaro_winkler (~> 1.5.1)
      parallel (~> 1.10)
      parser (>= 2.5, != 2.5.1.1)
      powerpack (~> 0.1)
      rainbow (>= 2.2.2, < 4.0)
      ruby-progressbar (~> 1.7)
      unicode-display_width (~> 1.0, >= 1.0.1)
    rubocop-performance (1.0.0)
      rubocop (>= 0.58.0)
    ruby-debug-ide (0.6.1)
      rake (>= 0.8.1)
    ruby-progressbar (1.10.0)
    ruby_dep (1.5.0)
    rubyzip (1.2.1)
    safe_yaml (1.0.4)
    sass (3.5.7)
      sass-listen (~> 4.0.0)
    sass-listen (4.0.0)
      rb-fsevent (~> 0.9, >= 0.9.4)
      rb-inotify (~> 0.9, >= 0.9.7)
    sass-rails (5.0.7)
      railties (>= 4.0.0, < 6)
      sass (~> 3.1)
      sprockets (>= 2.8, < 4.0)
      sprockets-rails (>= 2.0, < 4.0)
      tilt (>= 1.1, < 3)
    selenium-webdriver (3.14.0)
      childprocess (~> 0.5)
      rubyzip (~> 1.2)
    simple_form (4.0.1)
      actionpack (>= 5.0)
      activemodel (>= 5.0)
    spring (2.0.2)
      activesupport (>= 4.2)
    sprockets (3.7.2)
      concurrent-ruby (~> 1.0)
      rack (> 1, < 3)
    sprockets-rails (3.2.1)
      actionpack (>= 4.0)
      activesupport (>= 4.0)
      sprockets (>= 3.0.0)
    thor (0.20.0)
    thread_safe (0.3.6)
    tilt (2.0.8)
    timecop (0.9.1)
    tzinfo (1.2.5)
      thread_safe (~> 0.1)
    uglifier (4.1.18)
      execjs (>= 0.3.0, < 3)
    unicode-display_width (1.4.0)
    web-console (3.6.2)
      actionview (>= 5.0)
      activemodel (>= 5.0)
      bindex (>= 0.4.0)
      railties (>= 5.0)
    webmock (3.4.2)
      addressable (>= 2.3.6)
      crack (>= 0.3.2)
      hashdiff
    webpacker (3.5.5)
      activesupport (>= 4.2)
      rack-proxy (>= 0.6.1)
      railties (>= 4.2)
    websocket-driver (0.7.0)
      websocket-extensions (>= 0.1.0)
    websocket-extensions (0.1.3)
    xpath (3.1.0)
      nokogiri (~> 1.8)

PLATFORMS
  ruby

DEPENDENCIES
  attr_extras
  bootsnap (>= 1.1.0)
  brakeman
  byebug
  capybara (>= 2.15)
  capybara-screenshot
  config
  database_cleaner
  debase
  doorkeeper
  doorkeeper-mongodb
  dotenv-rails
  draper
  factory_bot_rails
  faraday
  faraday_middleware
  jbuilder (~> 2.5)
  json_logic!
  kaminari-actionview
  kaminari-mongoid
  listen (>= 3.0.5, < 3.2)
  local_time
  minitest-focus
  minitest-test_profile
  mongoid (~> 7.0)
  mongoid_rails_migrations
  omniauth (~> 1.8)
  omniauth-google-oauth2
  puma (~> 3.11)
  rails (~> 5.2.1)
  rails-controller-testing
  rspec-mocks
  rubocop
  rubocop-performance
  ruby-debug-ide
  sass-rails (~> 5.0)
  selenium-webdriver
  simple_form
  spring
  spring-watcher-listen!
  timecop
  tzinfo-data
  uglifier (>= 1.3.0)
  web-console (>= 3.3.0)
  webmock
  webpacker (~> 3.5)

RUBY VERSION
   ruby 2.5.3p105

BUNDLED WITH
   1.17.1
```

**Comment by johnsinco** [comment]
Have same issue.  Seems this codebase is no longer maintained.  If anyone has a new fork or replacement that's maintained would love to hear about it!  ;)

**Comment by nbulaj** [comment]
Would be added with a new version (5.2) after merge of #42

**Comment by nbulaj** [comment]
Released to rubygems as 5.2.0, feel free to report issues and PR's to fix them :)
Thanks! :tada:

**Comment by gsmendoza** [comment]
Thank you @nbulaj ! Will test this once we have the time. Feel free to close this issue :)
